### DISCUSSION TITLE
How to resolve this problem

### DISCUSSION THREAD

**Discussion by aafnnp** [vercel/next.js#37498] [Help] [answered]
```
pages/_app.tsx:48:28 - error TS2786: 'Component' cannot be used as a JSX component.
  Its element type 'ReactElement<any, any> | Component<{}, any, any> | null' is not a valid JSX element.
    Type 'Component<{}, any, any>' is not assignable to type 'Element | ElementClass | null'.
      Type 'Component<{}, any, any>' is not assignable to type 'ElementClass'.
        The types returned by 'render()' are incompatible between these types.
          Type 'React.ReactNode' is not assignable to type 'import("/Users/pfan/workspaces/transsnet/wallet-h5/node_modules/@types/react-dom/node_modules/@types/react/index").ReactNode'.
            Type '{}' is not assignable to type 'ReactNode'.
              Type '{}' is missing the following properties from type 'ReactPortal': key, children, type, props

48    <div className="page">{<Component {...pageProps} />}</div>

```

![image](https://user-images.githubusercontent.com/12791216/172307441-98db75a8-424e-41d4-9d79-5d6c55e33f0e.png)
![image](https://user-images.githubusercontent.com/12791216/172307474-92cebbcb-8e26-4caa-a535-d35bc4f1e466.png)

**Comment by icyJoseph** [comment, marked as answer]
Just do: `<div className="page"><Component {...pageProps} /></div>`, without the curly braces.

### Additionally

There seems to be a bug with types, described on this bug report, https://github.com/vercel/next.js/issues/37421 - Do not combine React 18 types with React 17 as it might lead to serious type unsafety.

**Reply by aafnnp** [reply]
Invalid

**Reply by icyJoseph** [reply]
Do you have a link to your repo? With this much info there's not much more I can do.

**Reply by aafnnp** [reply]
> Do you have a link to your repo? With this much info there's not much I can do.

sorry,it's company project.
i can provide more info but not upload to github.

**Reply by icyJoseph** [reply]
Looks like it's not needed, I've found a bug report, https://github.com/vercel/next.js/issues/37421 does it look like your issue? Don't change anything just yet.

**Reply by aafnnp** [reply]
i'll try.thx

**Reply by aafnnp** [reply]
works.thx

**Reply by icyJoseph** [reply]
Ah, no, but you shouldn't use React 18 types with React 17, unless you upgrade to React 18.... hope you are not mixing up type versions!

**Reply by aafnnp** [reply]
got it

**Reply by 0x80** [reply]
I have a similar issue when I tried to downgrade to react 17 in an app that was previously upgraded to 18 and nextjs 12. The error points to the react types inside of styled components:

```sh
yarn why @types/react
├─ @gemini/review-app@workspace:packages/review-app
│  └─ @types/react@npm:17.0.47 (via npm:^17)
│
...
│
└─ @types/styled-components@npm:5.1.25
   └─ @types/react@npm:18.0.9 (via npm:*)
```

I've tried to force resolutions to resolve to 17. I'm using yarn 3, but I can't get it to work. This `yarn set resolution @types/react@npm:18.0.9 17.0.47 --save` doesn't seem to do anything.

I've also tried `yarn set resolution @types/react@npm:^18 17.0.47 --save`

Any idea how to get resolutions to work in this case?

**Reply by aafnnp** [reply]
u should update `next,react,react-dom,@types/react,@types/react-dom`to latest

**Reply by 0x80** [reply]
I don't want to update them to latest. I'm trying to use react 17 instead of 18. It should be compatible with nextjs 12 afaik.

I'm trying to solve an issue that could be related to mobx, which has "poor man's support" for react 18 at the moment.

**Reply by aafnnp** [reply]
i have tried some methods.but only upgrade.
if u find ways to fixed.pls tell .thx

**Reply by 0x80** [reply]
Just for the record: I'm giving up on trying to use Next 12 with React 17. I have solved the issues I was facing via refactoring. The bug/conflict might still be there but it's not a problem anymore.

**Reply by icyJoseph** [reply]
```
"resolutions": {
    "**/@types/react": "^17.0.21",
    "**/@types/react-dom": "^17.0.9"
}
```

Perhaps this on the `package.json` should work out? Some other folks have been able to use Next 12 and React 17 by doing it.
